### PR TITLE
Added the ability to mark files as spoilers

### PR DESCRIPTION
**Pull request by davfsa**
### Checklist

- [x] I have run `nox` and all the pipelines have passed.
- [x] I have made unittests according to the code I have added/modified/deleted.

### CODE REVIEW

File: hikari/files.py [hunk 1]
```diff
@@ -817,22 +817,27 @@ class File(Resource[FileReader]):
...
     path: pathlib.Path
     """The path to the file."""
 
+    is_spoiler: bool
+    """Whether the file will be marked as a spoiler."""
+
     _filename: typing.Optional[str]
 
-    def __init__(self, path: Pathish, filename: typing.Optional[str] = None) -> None:
+    def __init__(self, path: Pathish, filename: typing.Optional[str] = None, spoiler: bool = False) -> None:
```
Comment (nekokatt) [inline hikari/files.py:837]:
```suggestion
    def __init__(self, path: Pathish, /, filename: typing.Optional[str] = None, *, spoiler: bool = False) -> None:
```

**Comment by nekokatt** [comment]
What about bytes, webresources/urls?

Can embeds have spoilt images? If so, we'd want to adjust the logic in the embed serializer to upload any spoilt resources again so that we can inject a file name, right?

**Comment by davfsa** [comment]
> What about bytes, webresources/urls?
Will add it to `Bytes`, websources and urls im not really sure can be spoilt (since they are text). Maybe adding them between `||` if the user selects to spoil them?

> Can embeds have spoilt images? If so, we'd want to adjust the logic in the embed serializer to upload any spoilt resources again so that we can inject a file name, right?
No, looks like they cant, they just render them as if they were not spoilers

File: hikari/files.py (hunk 1 above)
Comment (davfsa) [reply hikari/files.py]: Fixed :+1:

**Pull request merged**